src/arch/simddetect.cpp

```diff
@@ -210,15 +212,18 @@ SIMDDetect::SIMDDetect() {
 #endif
 
 #if defined(HAVE_NEON) && !defined(__aarch64__)
-#  ifdef ANDROID
+#  if HAVE_ANDROID_GETCPUFAMILY
```

Comment:
```suggestion
#  if defined(HAVE_ANDROID_GETCPUFAMILY)
```

Reply:
`HAVE_ANDROID_GETCPUFAMILY` is not always defined, so a simple `#if HAVE_ANDROID_GETCPUFAMILY` could trigger compiler warnings.

Reply:
An undefined macro evaluates to `0` in a `#if` preprocessor test.  No warning is emitted in this case.

Reply:
The GNU C++ compiler and clang emit warnings (depending on warning level). See https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html (-Wundef).